src/fitnesse/reporting/SuiteHtmlFormatter.java

```diff
@@ -177,6 +215,79 @@ public void testSystemStarted(TestSystem testSystem) {
       writeData(insertScript.html());
     }
   }
+  
+
+  
+  protected void showAssertionResult(Assertion assertion, String testResult, boolean overwriteLastMessage) {
+    if(testSystemListenerShowInstructions >0){
+      try {
+        String instructionText;
+        String position;
+        Expectation expectation = assertion.getExpectation();
+        if (expectation instanceof TableCell) {
+          TableCell cell = (TableCell) expectation;
+          position = String.format("% 3d,% 3d", cell.getRow(), cell.getCol());
+        } else {
+          position = "       ";
+        }
+        assertion.getInstruction();
+        instructionText = assertion.getInstruction().toString();
+        instructionText = position + ": " + instructionText + "-->" + testResult;
+        instructionText = HtmlUtil.escapeHTML(instructionText);
+        if (overwriteLastMessage){
+          this.instructionsHtmlText.removeLast();
+        } else {
+          this.instructionsHtmlText.removeFirst();
+        }        
+        this.instructionsHtmlText.addLast(instructionText);
+        String allInstructionsText = this.instructionsHtmlText.stream().collect(Collectors.joining("<br>"));
+        String html = "<h4>"+allInstructionsText+"</h4>";
+        String insertScript = JavascriptUtil.makeReplaceElementScript("step-by-step-Id2", html).html();
+        writeData(insertScript);
+      } catch (Exception e){
+        String html = e.getMessage();
+      }
+    }
+    if(testSystemListenerShowHtmlTable){
+      try{
+        Expectation expectation = assertion.getExpectation();
+        if (expectation instanceof TableCell) {
+          TableCell cell = (TableCell) expectation;
+          Table t = cell.getTable();
+          if (t instanceof HtmlTable) {
+            HtmlTable ht = (HtmlTable) t;
+            String html = ht.getTableNode().toHtml().toString();
+            String insertScript = JavascriptUtil.makeReplaceElementScript("step-by-step-Id", html).html();
+            testOutputChunk(null, insertScript);
+            String expandScript = JavascriptUtil.expandCurrentRow("step-by-step-Id").html();
+            testOutputChunk(null, expandScript);
+          }
+        }
+        if (testSystemListenerSleep > 0)
+          Thread.sleep(testSystemListenerSleep);
+      } catch (Exception e){
+        String em = e.getMessage();
+      }
+    }
+  }
+  
+  @Override
+  public void testAssertionVerified(Assertion assertion, TestResult testResult) {
```

Comment:
I think a unit test checking that `testAssertionVerified`, `testExceptionOccurred` and `testAssertionWillBeExecuted` all call `showAssertionResult()` (with the right parameters) would be good already. The actual HTML and JavaScript created don't have to be tested yet